pkg/private/tar/build_tar.py

```diff
@@ -31,27 +31,40 @@ class TarFile(object):
   class DebError(Exception):
     pass
 
-  def __init__(self, output, directory, compression, compressor, root_directory,
-               default_mtime):
-    self.directory = directory
+  def __init__(self, output, directory, compression, compressor, default_mtime):
+    # Directory prefix on all output paths
+    d = directory.strip('/')
+    self.directory = (d + '/') if d else None
     self.output = output
     self.compression = compression
     self.compressor = compressor
-    self.root_directory = root_directory
     self.default_mtime = default_mtime
 
   def __enter__(self):
     self.tarfile = archive.TarFileWriter(
         self.output,
         self.compression,
         self.compressor,
-        self.root_directory,
+        root_directory = self.directory,
         default_mtime=self.default_mtime)
     return self
 
   def __exit__(self, t, v, traceback):
     self.tarfile.close()
 
+  def normalize_path(self, path: str) -> str:
+    # Note: normpath changes / to \ on windows, but tarfile needs / style paths.
+    dest = os.path.normpath(path).replace(os.path.sep, '/')
+    # paths should not have a leading ./
+    if dest.startswith('./'):
```

Comment:
Should we use `for` instead of `if`?

Reply:
I don't understand.  How would a 'for' help here?

Reply:
Just being curious why we use 'for' to strip prepending `./` in the util functions but not here.

Reply:
It's just a question of do we have multiple things or not.
Here we have one path, so there is no need to iterate over a set.